# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/nsIQuotaManagerService.idl
# Commit: 396d9243d417
# Full Hash: 396d9243d4172aef258b24a09cf2a2574d86f87a
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-09 09:46:37
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Add a dedicated method for clearing storages of a specific client; r=dom-storage-reviewers,jari
#   
#   Client storages can be currently cleared by using the optional argument of
#   nsIQuotaManagerService::ClearStoragesForPrincipal. However, supporting clearing
#   storages for both a principal and a client is not ideal for several reasons.
# ==============================================================================

diff -r fe4a050ac4da -r 396d9243d417 dom/quota/nsIQuotaManagerService.idl
--- a/dom/quota/nsIQuotaManagerService.idl	Wed Oct 09 05:34:35 2024 +0000
+++ b/dom/quota/nsIQuotaManagerService.idl	Wed Oct 09 05:34:36 2024 +0000
@@ -266,6 +266,29 @@
                             [optional] in AString aClientType);
 
   /**
+   * Removes all storages stored for the given client. The files may not be
+   * deleted immediately depending on prohibitive concurrent operations.
+   *
+   * @param aPrincipal
+   *        A principal for the origin whose storages are to be cleared.
+   * @param aClientType
+   *        A string that tells what client type of storages will be cleared.
+   * @param aPersistenceType
+   *        An optional string that tells what persistence type of storages
+   *        will be cleared.  If omitted (or void), all persistence types will
+   *        be cleared for the principal and client type.  If a single
+   *        persistence type ("persistent", "temporary", or "default") is
+   *        provided, then only that persistence directory will be considered.
+   *        Note that "persistent" is different than being "persisted" via
+   *        persist() and is only for chrome principals.  See bug 1354500 for
+   *        more info.  In general, null is the right thing to pass here.
+   */
+  [must_use] nsIQuotaRequest
+  clearStoragesForClient(in nsIPrincipal aPrincipal,
+                         in AString aClientType,
+                         [optional] in ACString aPersistenceType);
+
+  /**
    * Removes all storages stored for the given prefix. The files may not be
    * deleted immediately depending on prohibitive concurrent operations.
    *
